Fixed the accessibility of WC_Cart::is_tax_displayed()

ObliviousHarmony opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

When we deprecated `WC_Legacy_Cart::tax_display_cart` in #26400 we introduced a fatal exception caused by the access modifier of `WC_Cart::is_tax_displayed()`. I've changed the access modifier on the method so that the parent class will be able to call it as well as users who have been directed to use it instead of the public property.

I discovered this because Subscriptions seems to be using the deprecated property.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

ObliviousHarmony committed:
Fixed the accessibility of WC_Cart::is_tax_displayed()

Since we're calling the function from the base class we need it to be protected or this throws a fatal exception.

helgatheviking commented:
Nice! Is it just me or are `is_tax_displayed()` and `display_prices_including_tax()` kind of redundant? They both check the customer and if the customer is VAT exempt.

ObliviousHarmony commented:
Howdy @helgatheviking,

The main difference is that `display_prices_including_tax` returns a boolean and is filtered while `is_tax_displayed()` returns `incl` or `excl` depending on configuration. An argument could be made that `is_tax_displayed()` isn't a very good name for this method though and perhaps we should change it since it was private before this PR.

helgatheviking commented:
Yes, I see the filter... not sure if that's good to have in my case or not. Guess we'll find out!

By redundant, I meant that `display_prices_including_tax()` checks the customer VAT status via 
`$customer_exempt = $this->get_customer() && $this->get_customer()->get_is_vat_exempt();` and then calls `is_tax_displayed()` which checks the customer VAT status again. So `display_prices_including_tax()` checks the VAT status twice.

codestor4 commented:
I think @helgatheviking makes a good point. To avoid repeating our code, we should probably change the following function:

```
public function display_prices_including_tax() {
		$customer_exempt = $this->get_customer() && $this->get_customer()->get_is_vat_exempt();

		return apply_filters( 'woocommerce_cart_' . __FUNCTION__, 'incl' === $this->is_tax_displayed() && ! $customer_exempt );
	}
```

into this:

```
public function display_prices_including_tax() {
		return apply_filters( 'woocommerce_cart_' . __FUNCTION__, 'incl' === $this->is_tax_displayed() );
	}
```

> An argument could be made that is_tax_displayed() isn't a very good name for this method though and perhaps we should change it since it was private before this PR.

I agree. I was a bit confused with the name. It'd be nice if we update `is_tax_displayed`  to something similar to `display_prices_including_tax` so it'd be easy to relate both functions (one returns a bool, another returns incl/excl).

@ObliviousHarmony what do you think?

claudiosanches reviewed (approved):
Thanks!

ObliviousHarmony committed:
Renamed `WC_Cart::is_tax_displayed()` to `WC_Cart::get_tax_price_display_mode()` to clarify intent

ObliviousHarmony commented:
@codestor4 @helgatheviking 

I see what you're getting at. I've gone ahead and removed the redundant code and renamed the method to make it a bit more clear.
